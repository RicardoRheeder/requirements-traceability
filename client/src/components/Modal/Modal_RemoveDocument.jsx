import React, { useState } from 'react'
import {
  deleteDocAsync,
  fetchUserDocsAsync,
} from '../../redux/stores/document/actions'
import { useDispatch, useSelector } from 'react-redux'
import { useAuth0 } from '@auth0/auth0-react'
import { setModalObject } from '../../redux/stores/common/actions'
import { fetchUserRecentDocsAsync } from '../../redux/stores/user/actions'

export default function Modal_RemoveDocument() {
  const { user } = useAuth0()
  const dispatch = useDispatch()
  const selectedDoc = useSelector(
    (state) => state.common.selectedDocumentPanelObject
  )
  const userInfo = useSelector((state) => state.user.info)

  const [doc, setDoc] = useState({
    title: '',
  })

  const handleSubmit = (e) => {
    e.preventDefaults
    if (selectedDoc.title.trim() === doc.title.trim()) {
      dispatch(deleteDocAsync({ id: selectedDoc._id, user: userInfo._id }))
    }
    setDoc({ title: '' })
    dispatch(setModalObject({ visible: false, mode: 0 }))
    dispatch(fetchUserDocsAsync(user))
    dispatch(fetchUserRecentDocsAsync(user.email))

  }

  const handleChange = (e) => {
    const { value } = e.target
    setDoc({ ...doc, title: value })
  }

  return (
    <div className="modal-root modal-root-child">
      <div className="modal-contents-container">
        <h1 className="modal-contents-title">Remove document</h1>
        <h2>
          For authorization, please enter the name of the document you want to
          permanently remove.
        </h2>
        <br></br>
        <h3>
          Please type{' '}
          <span className="remove-document-title-span">
            {selectedDoc.title}
          </span>{' '}
          below and click 'submit' to permanently delete the document.
        </h3>
        <form onSubmit={handleSubmit}>
          <input className="modal-input" onChange={handleChange} />

          <div className="button-container">
            <button className="orange-button modal-button">
              Submit
            </button>
            <button
              className="orange-button modal-button"
              onClick={() =>
                dispatch(setModalObject({ visible: false, mode: 0 }))
              }
            >
              Close
            </button>
          </div>
        </form>
      </div>
    </div>
  )
}
