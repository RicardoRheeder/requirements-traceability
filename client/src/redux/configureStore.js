import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
