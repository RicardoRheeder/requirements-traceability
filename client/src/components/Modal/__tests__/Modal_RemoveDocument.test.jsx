import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import Modal_RemoveDocument from "../Modal_RemoveDocument"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("Remove document modal component", () => {
    let store
    beforeEach(() => {
        store = mockStore({common:{
            modalObject: { visible: true},
            selectedDocumentPanelObject: {_id:"test"}
        },
        user: {info: "testinfo"}});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><Modal_RemoveDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><Modal_RemoveDocument/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});