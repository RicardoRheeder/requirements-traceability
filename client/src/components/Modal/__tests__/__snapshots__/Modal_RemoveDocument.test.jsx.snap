// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove document modal component renders 1`] = `ShallowWrapper {}`;

exports[`Remove document modal component renders with children 1`] = `ReactWrapper {}`;
